### PR TITLE
Math magicians: components

### PR DESCRIPTION
**Pull request by okoyecharles**
## Description
In this project, you should continue with the development of the Math Magicians app. You will develop a React component that will hold the core functionality: a calculator.

## Changes Made ♻️ 
- [x] Use React components.
- [x] Use React props.
- [x] Use React state.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,10 @@
-import logo from './logo.svg';
 import './App.css';
+import Calculator from './components/Calculator';
 
 function App() {
   return (
     <div className="App">
```
Comment (cwaku) [inline src/App.js]: - [ ] It is required that your `App.js` component is also class-based. Kindly change it from a functional component to a class-based one just as you did for the `Calculator.js component.

File: src/components/Calculator.css [hunk 2]
```diff
@@ -0,0 +1,64 @@
+.calculator__container {
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
+  gap: 0.125em;
+  width: max-content;
+  background: #d3d3d3;
+}
```
Comment (cwaku) [inline src/components/Calculator.css]: [OPTIONAL] 
- [ ] Your app is not quite responsive on some screen dimensions, kindly fix that to allow ease of use on more devices.

image

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @OkoyeCharles 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️  but you are almost there! 🏃🏾 

## Highlights
- [x] All liners passed 💯 
- [x] Good PR description 👍🏾 
- [x] Good implementation ✔️ 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them, in order to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (okoyecharles) [reply src/App.js]: thanks this has been done

File: src/components/Calculator.css (hunk 2 above)
Comment (okoyecharles) [reply src/components/Calculator.css]: This has bee done thank you

File: src/App.js [hunk 3]
```diff
@@ -1,27 +1,10 @@
-import logo from './logo.svg';
 import './App.css';
+import Calculator from './components/Calculator';
 
 function App() {
```
Comment (youmari) [inline src/App.js]: - [x] as the previous reviewer state you should change your `App`component from function to class-based so please try to do that in order to meet the requirements

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @OkoyeCharles , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✅ PRO PR
✅ Git flow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 3 above)
Comment (okoyecharles) [reply src/App.js]: @youmari  I'm sorry but what should I do I have already done that.. please explain what exactly I should do I have already converted the app function component to a class component.

Comment (okoyecharles) [reply src/App.js]: @youmari PLEASE view the changes made in the previous commit I made the page responsive and changed the app component to a class, as I only have one review left.

Comment (okoyecharles) [reply src/App.js]: @youmari and if there are any changes required also let me know in the comments.

Comment (youmari) [reply src/App.js]: @OkoyeCharles You should change the app function to class-based  
For example : 
```JS
class App extends React.Component {
}
```

Comment (okoyecharles) [reply src/App.js]: @youmari but that is what i did

Comment (okoyecharles) [reply src/App.js]: @youmari can we call on zoom so i could show you what i have done

Comment (okoyecharles) [reply src/App.js]: @youmari  below is my current App.js.
https://github.com/OkoyeCharles/math-magicians/blob/1deb486ca2ca5bba0febbd326b6a3f7fa5e0abd5/src/App.js#L1-L20

Please cross-check what branch you are on, you should be on the [calculator branch](https://github.com/OkoyeCharles/math-magicians/tree/calculator)

I'm sorry for giving you trouble but I a positively sure that I got that correct.

**Review by Kingstalux — APPROVED** [review]
Hi @OkoyeCharles,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966
 

Good job using class-based components for both your `App.js` and `Calculator.js` :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**